examples/bitmasked.py

```diff
@@ -0,0 +1,42 @@
+import taichi as ti
+ti.init()
+
+n = 256
+x = ti.var(ti.f32)
+# `bitmasked` is a tensor that supports sparsity, in that each element can be
+# activated individually. (It can be viewed as `dense`, with an extra bit for each
+# element to mark its activation). Assigning to an element will activate it
+# automatically. Use struct-for syntax to loop over the active elements only.
+ti.root.bitmasked(ti.ij, (n, n)).place(x)
+
+
+@ti.kernel
+def activate():
+    # All elements in bitmasked is initially deactivated
+    # Let's activate elements in the rectangle now!
+    for i, j in ti.ndrange((100, 125), (100, 125)):
+        x[i, j] = 0  # assign any value to activate the element at (i, j)
+
+
+@ti.kernel
+def paint_active_pixels(t: ti.f32):
+    # struct-for syntax: loop over active pixels, inactive pixels are excluded
+    for i, j in x:
+        x[i, j] = ti.sin(t)
```

Comment:
Because this can go negative, half of the time the graph is pure black. WDYT if we map the range from [-1, 1] to [0, 1]?